spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/listener/FifoSqsComponentFactory.java

```diff
@@ -70,9 +71,11 @@ public MessageSource<T> createMessageSource(SqsContainerOptions options) {
 	@Override
 	public MessageSink<T> createMessageSink(SqsContainerOptions options) {
 		MessageSink<T> deliverySink = createDeliverySink(options.getListenerMode());
-		return new MessageGroupingSinkAdapter<>(
-				maybeWrapWithVisibilityAdapter(deliverySink, options.getMessageVisibility()),
-				getMessageGroupingFunction());
+		MessageSink<T> wrappedDeliverySink = maybeWrapWithVisibilityAdapter(deliverySink,
+				options.getMessageVisibility());
+		return options.getMessageGrouping()
+				? new MessageGroupingSinkAdapter<>(wrappedDeliverySink, getMessageGroupingFunction())
+				: new MessageSinkAdapter<>(wrappedDeliverySink);
```

Comment:
Why do we need this `MessageSinkAdapter`?

Can't we return the `wrappedDeliverySink` directly?